### PR TITLE
Sync SVGAnimationElement.idl with the specs

### PR DESCRIPTION
**Pull request by Ahmad-S792**
<pre>
Sync SVGAnimationElement.idl with the specs
<a href="https://bugs.webkit.org/show_bug.cgi?id=250575">https://bugs.webkit.org/show_bug.cgi?id=250575</a>

Reviewed by NOBODY (OOPS!).

This patch is to align WebKit with Gecko / Firefox, Blink / Chromium and Web-Specifications.

Merge - <a href="https://src.chromium.org/viewvc/blink?view=rev&revision=171485">https://src.chromium.org/viewvc/blink?view=rev&revision=171485</a>

This patch aligns WebKit with web-specification by dropping unnecessary NaN (non-finite values) 
and remove 'optional unrestricted' from IDL files as needed across files.

This patch aligns WebKit with web-specification by dropping unnecessary
NaN (non-finite values) early return and remove 'optional unrestricted' from IDL files as needed across files.

* Source/WebCore/svg/SVGAnimationElement.cpp:
(SVGAnimationElement::beginElementAt): Remove early return for 'nan' values
(SVGAnimationElement::endElementAt): Ditto
* Source/WebCore/svg/SVGAnimationElement.idl: Align with web-spec
* LayoutTests/svg/dom/method-argument-typechecks.html: Add Test Case
* LayoutTests/svg/dom/method-argument-typechecks-expected.txt: Add Test Case Expectation
* LayoutTests/svg/dom/method-argument-aritychecks.html: Add Test Case
* LayoutTests/svg/dom/method-argument-aritychecks-expected.txt: Add Test Case Expectation
* LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/idlharness.any-expected.txt: Rebaselined
* LayoutTests/svg/custom/elementTimeControl-nan-crash-expected.txt: Ditto
* LayoutTests/platform/glib/svg/custom/elementTimeControl-nan-crash-expected.txt: Add Platform Specific Expectation
* LayoutTests/platform/ios/svg/custom/elementTimeControl-nan-crash-expected.txt: Add Platform Specific Expectation
* LayoutTests/platform/win/svg/custom/elementTimeControl-nan-crash-expected.txt: Add Platform Specific Expectation
</pre>

https://github.com/WebKit/WebKit/commit/02e312fd9d71c024c5a0848ce8f305451723f5d8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105678 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14769 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38553 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114906 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/175047 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109580 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16175 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5968 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97945 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114714 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111432 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12305 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95287 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39784 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/94169 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26927 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8037 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28280 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8533 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4865 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14154 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47827 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10086 "Built successfully") | | | 
| | | | |

### CODE REVIEW

**Comment by weinig** [comment]
Would be great to make sure there is WPT coverage for this.

File: Source/WebCore/svg/SVGAnimationElement.cpp [hunk 1]
```diff
@@ -257,7 +258,7 @@ void SVGAnimationElement::beginElement()
 
 void SVGAnimationElement::beginElementAt(float offset)
 {
-    if (std::isnan(offset))
+    if (std::isfinite(offset))
         return;
```
Comment (shallawa) [inline Source/WebCore/svg/SVGAnimationElement.cpp]: I think the check here should be the opposite:

```
  if (!std::isfinite(offset))
        return;
```

File: Source/WebCore/svg/SVGAnimationElement.cpp [hunk 2]
```diff
@@ -270,7 +271,7 @@ void SVGAnimationElement::endElement()
 
 void SVGAnimationElement::endElementAt(float offset)
 {
-    if (std::isnan(offset))
+    if (std::isfinite(offset))
         return;
```
Comment (shallawa) [inline Source/WebCore/svg/SVGAnimationElement.cpp]: Ditto.

File: Source/WebCore/svg/SVGAnimationElement.cpp (hunk 1 above)
Comment (cdumez) [reply Source/WebCore/svg/SVGAnimationElement.cpp]: Given the current IDL, offset should never be infinite (or we would have thrown on bindings side). So while I agree with Said that the check seems backwards, I think we shouldn't even have a check unless this code has other call sites besides the bindings.

Comment (Ahmad-S792) [reply Source/WebCore/svg/SVGAnimationElement.cpp]: I opted to remove and it seems to work fine and I have rebaselined all tests now. Thanks for feedback.

File: Source/WebCore/svg/SVGAnimationElement.cpp (hunk 2 above)
Comment (Ahmad-S792) [reply Source/WebCore/svg/SVGAnimationElement.cpp]: Done.

**Comment by Ahmad-S792** [comment]
> Would be great to make sure there is WPT coverage for this.

Covered via IDLharness testcase. Just rebaslined and have two progressions. :-)

**Comment by shallawa** [comment]
Please make this PR for fixing the SVGAnimationElement.idl only as the title says. I think you need to remove the changes in SVGFEDropShadowElement and SVGFEGaussianBlurElement. These changes are addressed by https://github.com/WebKit/WebKit/pull/9159

**Comment by Ahmad-S792** [comment]
> Please make this PR for fixing the SVGAnimationElement.idl only as the title says. I think you need to remove the changes in SVGFEDropShadowElement and SVGFEGaussianBlurElement. These changes are addressed by #9159

Done now. :-)

File: LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/idlharness.any-expected.txt [hunk 3]
```diff
@@ -320,7 +320,7 @@ PASS SVGFEDropShadowElement interface: attribute dx
 PASS SVGFEDropShadowElement interface: attribute dy
 PASS SVGFEDropShadowElement interface: attribute stdDeviationX
 PASS SVGFEDropShadowElement interface: attribute stdDeviationY
-FAIL SVGFEDropShadowElement interface: operation setStdDeviation(float, float) assert_equals: property has wrong .length expected 2 but got 0
+PASS SVGFEDropShadowElement interface: operation setStdDeviation(float, float)
```
Comment (shallawa) [inline LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/idlharness.any-expected.txt:323]: This part is not related to this PR.

File: LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/idlharness.any-expected.txt [hunk 4]
```diff
@@ -355,7 +355,7 @@ PASS SVGFEGaussianBlurElement interface: attribute in1
 PASS SVGFEGaussianBlurElement interface: attribute stdDeviationX
 PASS SVGFEGaussianBlurElement interface: attribute stdDeviationY
 PASS SVGFEGaussianBlurElement interface: attribute edgeMode
-FAIL SVGFEGaussianBlurElement interface: operation setStdDeviation(float, float) assert_equals: property has wrong .length expected 2 but got 0
+PASS SVGFEGaussianBlurElement interface: operation setStdDeviation(float, float)
```
Comment (shallawa) [inline LayoutTests/imported/w3c/web-platform-tests/css/filter-effects/idlharness.any-expected.txt:358]: Ditto.

File: LayoutTests/platform/win/svg/custom/elementTimeControl-nan-crash-expected.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+CONSOLE MESSAGE: TypeError: The provided value is non-finite
```
Comment (shallawa) [inline LayoutTests/platform/win/svg/custom/elementTimeControl-nan-crash-expected.txt:1]: Ditto.

File: LayoutTests/svg/custom/elementTimeControl-nan-crash-expected.txt [hunk 6]
```diff
@@ -1 +1,11 @@
-Test for WK100322: ElementTimeControl should check for invalid values. This test passes if it does not crash.
+CONSOLE MESSAGE: TypeError: The provided value is non-finite
```
Comment (shallawa) [inline LayoutTests/svg/custom/elementTimeControl-nan-crash-expected.txt:1]: Why do you want to make the expected file a dump render tree instead of dump as text? Also I do not see the changes in the test file `LayoutTests/svg/custom/elementTimeControl-nan-crash.html` to make it output this result.

File: LayoutTests/svg/dom/method-argument-aritychecks.html [hunk 7]
```diff
@@ -0,0 +1,51 @@
+<!DOCTYPE html>
```
Comment (shallawa) [inline LayoutTests/svg/dom/method-argument-aritychecks.html:1]: Please choose a more descriptive name for the test like `SVGAnimationElement-begin-end-arguments.html` or something similar.

File: LayoutTests/svg/dom/method-argument-aritychecks.html [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+debug('');
+debug('');
+debug('SVGFEGaussianBlurElement');
+
+debug('');
+debug('setStdDeviation(float stdDeviationX, float stdDeviationY)');
+shouldThrow('gaussian.setStdDeviation()');
+shouldThrow('gaussian.setStdDeviation(0)');
+shouldNotThrow('gaussian.setStdDeviation(0, 0)');
```
Comment (shallawa) [inline LayoutTests/svg/dom/method-argument-aritychecks.html:49]: Please remove.

File: LayoutTests/svg/dom/method-argument-typechecks-expected.txt [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+
+SVGFEGaussianBlurElement
+
+setStdDeviation(float stdDeviationX, float stdDeviationY)
+PASS gaussian.setStdDeviation(0, 0) did not throw exception.
+PASS gaussian.setStdDeviation(NaN, 1) threw exception TypeError: The provided value is non-finite.
+PASS gaussian.setStdDeviation(Infinity, 1) threw exception TypeError: The provided value is non-finite.
+PASS gaussian.setStdDeviation(1, NaN) threw exception TypeError: The provided value is non-finite.
+PASS gaussian.setStdDeviation(1, Infinity) threw exception TypeError: The provided value is non-finite.
+PASS successfullyParsed is true
```
Comment (shallawa) [inline LayoutTests/svg/dom/method-argument-typechecks-expected.txt:37]: Please remove.

File: LayoutTests/svg/dom/method-argument-typechecks.html [hunk 10]
```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
```
Comment (shallawa) [inline LayoutTests/svg/dom/method-argument-typechecks.html:1]: Please choose a more descriptive name for the test.

File: LayoutTests/svg/dom/method-argument-typechecks.html [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+debug('');
+debug('SVGFEGaussianBlurElement');
+
+debug('');
+debug('setStdDeviation(float stdDeviationX, float stdDeviationY)');
+shouldNotThrow('gaussian.setStdDeviation(0, 0)');
+shouldThrow('gaussian.setStdDeviation(NaN, 1)');
+shouldThrow('gaussian.setStdDeviation(Infinity, 1)');
+shouldThrow('gaussian.setStdDeviation(1, NaN)');
+shouldThrow('gaussian.setStdDeviation(1, Infinity)');
```
Comment (shallawa) [inline LayoutTests/svg/dom/method-argument-typechecks.html:51]: Please remove.

File: LayoutTests/svg/dom/method-argument-aritychecks-expected.txt [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+PASS dropShadow.setStdDeviation(0, 0) did not throw exception.
+
+
+SVGFEGaussianBlurElement
+
+setStdDeviation(float stdDeviationX, float stdDeviationY)
+PASS gaussian.setStdDeviation() threw exception TypeError: Not enough arguments.
+PASS gaussian.setStdDeviation(0) threw exception TypeError: Not enough arguments.
+PASS gaussian.setStdDeviation(0, 0) did not throw exception.
+PASS successfullyParsed is true
```
Comment (shallawa) [inline LayoutTests/svg/dom/method-argument-aritychecks-expected.txt:31]: This changes are not related to this PR. Please remove.
